File: 0101-0200/0150-Evaluate Reverse Polish Notation/0150-Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> St;
        for (string s : tokens) {
            if (s == "+" || s == "-" || s == "*" || s == "/") {
                int val2 = St.top();
                St.pop();
                int val1 = St.top();
                St.pop();
                if (s == "+") {
                    St.push(val1 + val2);
                }
                else if (s == "-") {
                    St.push(val1 - val2);
                }
                else if (s == "*") {
                    St.push(val1 * val2);
                }
                else {
                    St.push(val1 / val2);
                }
            }
            else {
                St.push(stoi(s));
            }
        }
        
        return St.top();
    }
};

File: 0101-0200/0150-Evaluate Reverse Polish Notation/0150-Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        St = []
        for token in tokens:
            if token.isdigit() or len(token) > 1:
                St.append(int(token))
            else:
                b = St.pop()
                a = St.pop()
                if token == '+':
                    St.append(a + b)
                elif token == '-':
                    St.append(a - b)
                elif token == '*':
                    St.append(a * b)
                else:
                    St.append(int(a / b))
        return St[-1]
